src/append_only_data.rs

```diff
@@ -271,76 +329,77 @@ struct AppendOnly<P: Perm> {
-    // could represent an individual user, or a group of users, depending on the `PublicKey` type.
+    /// This is the history of owners, with each entry representing an owner. Each single owner
+    /// could represent an individual user, or a group of users, depending on the `PublicKey` type.
     owners: Vec<Owner>,
 }
 
 /// Common methods for all `AppendOnlyData` flavours.
 pub trait AppendOnlyData<P> {
-    /// Return a value for the given key (if it is present).
+    /// Returns a value for the given key, if present.
     fn get(&self, key: &[u8]) -> Option<&Vec<u8>>;
 
-    /// Return the last entry in the Data (if it is present).
+    /// Returns the last entry, if present.
     fn last_entry(&self) -> Option<&Entry>;
 
-    /// Get a list of keys and values with the given indices.
+    /// Gets a list of keys and values with the given indices.
     fn in_range(&self, start: Index, end: Index) -> Option<Entries>;
 
-    /// Return all entries.
+    /// Returns all entries.
     fn entries(&self) -> &Entries;
 
-    /// Return the address of this AppendOnlyData.
+    /// Returns the address.
     fn address(&self) -> &Address;
 
-    /// Return the name of this AppendOnlyData.
+    /// Returns the name.
     fn name(&self) -> &XorName;
 
-    /// Return the type tag of this AppendOnlyData.
+    /// Returns the type tag.
     fn tag(&self) -> u64;
 
-    /// Return the last entry index.
+    /// Returns the last entry index.
     fn entries_index(&self) -> u64;
 
-    /// Return the last owners index.
+    /// Returns the last owners index.
     fn owners_index(&self) -> u64;
 
-    /// Return the last permissions index.
+    /// Returns the last permissions index.
     fn permissions_index(&self) -> u64;
 
-    /// Get a complete list of permissions from the entry in the permissions list at the specified
+    /// Gets a complete list of permissions from the entry in the permissions list at the specified
     /// index.
     fn permissions_range(&self, start: Index, end: Index) -> Option<&[P]>;
 
-    /// Add a new permissions entry.
+    /// Adds a new permissions entry.
     /// The `Perm` struct should contain valid indices.
-    fn append_permissions(&mut self, permissions: P, permissions_idx: u64) -> Result<()>;
+    fn append_permissions(&mut self, permissions: P, permissions_index: u64) -> Result<()>;
 
-    /// Fetch perms at index.
+    /// Fetches permissions at index.
     fn permissions(&self, perm_index: impl Into<Index>) -> Option<&P>;
 
-    /// Fetch owner at index.
+    /// Fetches owner at index.
     fn owner(&self, owners_index: impl Into<Index>) -> Option<&Owner>;
 
-    /// Get a complete list of owners from the entry in the permissions list at the specified index.
+    /// Gets a complete list of owners from the entry in the permissions list at the specified
+    /// index.
     fn owners_range(&self, start: Index, end: Index) -> Option<&[Owner]>;
 
-    /// Add a new owner entry.
+    /// Adds a new owner entry.
```

Comment:
Worth expanding on what `owners_index` means in this context (should be equal to `current owners index + 1`)
Something like
```rust
/// If the specified `owners_index` does not match the last recorded owners index + 1, an
/// error will be returned.
```